adr/2024_10_22_filters_structure_refactor.md

```diff
@@ -0,0 +1,69 @@
+# API - Filters Structure Refactor
+
+* Status: Accepted
+* Date: 2024-10-22
+
+In effect from version: 2.0
+
+Related PRs: #17290
+
+## Context and Problem Statement
+
+In previous versions, Sylius had filters grouped by field type rather than by resource and section. This led to filters being applied inconsistently, making it hard to customize them per resource. The filters were also located in the `Filter/Doctrine/` directory, which didn't align with how we organize other Doctrine-related features, creating confusion.
```

Comment:
```suggestion
In previous versions, Sylius had API filters grouped by field type rather than by resource and section. This led to filters being applied inconsistently, making it hard to customize them per resource. The filters were also located in the `Filter/Doctrine/` directory, which didn't align with how we organize other Doctrine-related features, creating confusion.
```